unittest/server/src/ivi_controller_uinttests.cpp

```diff
@@ -0,0 +1,3475 @@
+    uint32_t l_id = 10;
+    controller_screen_screenshot(nullptr, nullptr, nullptr, l_id);
+
+    ASSERT_EQ(wl_resource_post_no_memory_fake.call_count, 0);
+    ASSERT_EQ(wl_resource_create_fake.call_count, 1);
+    ASSERT_EQ(weston_buffer_from_resource_fake.call_count, 1);
+    ASSERT_EQ(wl_resource_post_event_fake.call_count, 1);
+    ASSERT_EQ(wl_resource_post_event_fake.arg1_history[0], IVI_SCREENSHOT_ERROR);
+    ASSERT_EQ(wl_shm_buffer_get_data_fake.call_count, 0);
+}
+
+/** ================================================================================================
+ * @test_id             controller_screen_screenshot_success
+ * @brief               Test case of controller_screen_screenshot() where wl_resource_create() success, return an object
+ *                      and wl_resource_get_user_data() success, return an object
+ * @test_procedure Steps:
+ *                      -# Mocking the wl_resource_create() does return an object
+ *                      -# Mocking the wl_resource_get_user_data() does return an object
+ *                      -# Calling the controller_screen_screenshot()
+ *                      -# Verification point:
+ *                         +# wl_resource_set_implementation() must be called once time
+ *                         +# wl_list_insert() must be called 2 times
+ *                         +# Free resources are allocated when running the test
+ */
+TEST_F(ControllerTests, controller_screen_screenshot_success)
+{
+    struct wl_resource * screenshot[1] = {(struct wl_resource *)0xFFFFFFFF};
+    SET_RETURN_SEQ(wl_resource_create, screenshot, 1);
+    SET_RETURN_SEQ(wl_resource_get_user_data, (void**)mp_iviScreen, 1);
+    SET_RETURN_SEQ(weston_buffer_from_resource, &mp_westonBuffer, 1);
+
+    uint32_t l_id = 10;
+    controller_screen_screenshot(nullptr, nullptr, nullptr, l_id);
+
+    ASSERT_EQ(wl_resource_post_no_memory_fake.call_count, 0);
+    ASSERT_EQ(wl_resource_create_fake.call_count, 1);
+    ASSERT_EQ(weston_buffer_from_resource_fake.call_count, 1);
+    ASSERT_EQ(wl_resource_destroy_fake.call_count, 0);
+
+    struct screenshot_frame_listener *lp_l = (struct screenshot_frame_listener*)wl_resource_set_implementation_fake.arg2_history[0];
+    free(lp_l);
+}
+
+/** ================================================================================================
+ * @test_id             controller_screen_get_nullScreen
+ * @brief               Test case of controller_screen_get() where wl_resource_get_user_data() fails, return null pointer
+ * @test_procedure Steps:
+ *                      -# Calling the controller_screen_get()
+ *                      -# Verification point:
+ *                         +# wl_resource_post_event() must be called once time
+ */
+TEST_F(ControllerTests, controller_screen_get_nullScreen)
+{
+    uint32_t l_param = 1;
+    controller_screen_get(nullptr, nullptr, l_param);
+
+    ASSERT_EQ(wl_resource_post_event_fake.call_count, 1);
```

Comment:
should verify wl_resource_post_event_fake.arg1_history and get_layers_on_screen_fake.call_count